actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -212,10 +211,10 @@ def no_routes(routes)
             <<~MESSAGE
             You don't have any routes defined!
 
-            Please add some routes in config/routes.rb.\n
+            Please add some routes in config/routes.rb.
             MESSAGE
           else
-            "No routes were found for this controller\n"
+            "No routes were found."
```

Comment:
Perhaps we should tune this to `No routes found matching conditions.` to indicate to the user that their `-g` or `-c` was applied.

Reply:
@kaspth What do you think about displaying a message depending on which option was passed `-c` or `-g`.

```diff
diff --git a/actionpack/lib/action_dispatch/routing/inspector.rb b/actionpack/lib/action_dispatch/routing/inspector.rb
index 73f0531..36128b3 100644
--- a/actionpack/lib/action_dispatch/routing/inspector.rb
+++ b/actionpack/lib/action_dispatch/routing/inspector.rb
@@ -64,7 +64,7 @@ def format(formatter, filter = nil)
         routes_to_display = filter_routes(normalize_filter(filter))
         routes = collect_routes(routes_to_display)
         if routes.none?
-          formatter.no_routes(collect_routes(@routes))
+          formatter.no_routes(collect_routes(@routes), filter)
           return formatter.result
         end
 
@@ -82,9 +82,9 @@ def format(formatter, filter = nil)
       private
 
         def normalize_filter(filter)
-          if filter.is_a?(Hash) && filter[:controller]
+          if filter[:controller]
             { controller: /#{filter[:controller].downcase.sub(/_?controller\z/, '').sub('::', '/')}/ }
-          elsif filter
+          elsif filter[:grep_pattern]
             { controller: /#{filter}/, action: /#{filter}/, verb: /#{filter}/, name: /#{filter}/, path: /#{filter}/ }
           end
         end
@@ -147,7 +147,7 @@ def header(routes)
           @buffer << draw_header(routes)
         end
 
-        def no_routes(routes)
+        def no_routes(routes, filter)
           @buffer <<
           if routes.none?
             <<~MESSAGE
@@ -205,7 +205,7 @@ def section(routes)
         def header(routes)
         end
 
-        def no_routes(routes)
+        def no_routes(routes, filter)
           @buffer <<
           if routes.none?
             <<~MESSAGE
@@ -213,8 +213,10 @@ def no_routes(routes)
 
             Please add some routes in config/routes.rb.
             MESSAGE
-          else
-            "No routes were found."
+          elsif filter.has_key?(:controller)
+            "No routes were found for this controller."
+          elsif filter.has_key?(:grep_pattern)
+            "No routes were found for this pattern."
           end
           @buffer << "For more information about routes, see the Rails guide: http://guides.rubyonrails.org/routing.html."
         end
diff --git a/railties/lib/rails/commands/routes/routes_command.rb b/railties/lib/rails/commands/routes/routes_command.rb
index c4f3717..859ab9c 100644
--- a/railties/lib/rails/commands/routes/routes_command.rb
+++ b/railties/lib/rails/commands/routes/routes_command.rb
@@ -39,7 +39,7 @@ def routes_filter
           if options.has_key?("controller")
             { controller: options["controller"] }
           elsif options.has_key?("grep_pattern")
-            options["grep_pattern"]
+            { grep_pattern: options["grep_pattern"] }
           else
             nil
           end
```